scripts/ci-build.sh

```diff
@@ -0,0 +1,562 @@
+      d="$PWD"
+      cd $name || echo "Layer name wrong?  Can't cd to dir : $name"
+      git fetch --all
+
+      # If forked, make sure to use the right remote, otherwise origin
+      # For branches, prefer the explicit: remote/branchname
+      if git remote | grep -q temp_fork ; then
+        git checkout "temp_fork/$value" 2>/dev/null || git checkout $value
+      else
+        git checkout "origin/$value" 2>/dev/null || git checkout $value
+      fi
+
+      cd "$d"
+      break  # First one wins, priority order: COMMIT >= TAG >= BRANCH
+    fi
+  done
+}
+
+# Overrides are defined invisibly in environment variables so we should inform
+# the user which ones are applied, for both interactive use and for studying
+# a build log at a later time.
+# NOTE:  Because of the order they are called we can't use output from the
+# layer_override function above, so we have some repeated code here.
+print_layer_overrides() {
+  local names="$1" name clean_name suffix var_name value d
+
+  for name in $names ; do
+    clean_name=$(echo "$name" | sed 's/-/_/g')
+    for suffix in COMMIT TAG BRANCH ; do
+      var_name="LAYER_${clean_name}_${suffix}"
+      value=$(deref "$var_name")
+      if [ -n "$value" ] ; then
+        echo "$var_name = $value"
+      fi
+    done
+  done
+}
+
+# List all dependent layers here.  This list is used by the layer overrides
+# functionality.
+# NOTE If a super-project is applied, such as using submodules, then this list
+# would be fetched from git instead, in case it changes.
+LAYERS="
+poky
+meta-gplv2
+meta-openembedded
+renesas-rcar-gen3
+"
+
+# Clean up function is called at end of script, or as a signal handler (trap)
+# if the script is interrupted
+cleanup() {
+  # Restore git config user - if this was not defined locally before then it is
+  # unset (which might mean a global setting is used)
+  if [ -z "$olduser" ] ; then
+    git config --unset user.name
+  else
+    git config user.name "$olduser"
+  fi
+  if [ -z "$oldemail" ] ; then
+    git config --unset user.email
+  else
+    git config user.email "$oldemail"
+  fi
+}
+
+# ---- Main program ----
+
+trap cleanup SIGINT SIGTERM
+
+D=$(dirname "$0")
+cd "$D"
+BASEDIR=$(readlink -f "$PWD/..")
+
+# The user must define this
+ensure_var_is_defined TARGET
+
+stop_if_failure
+
+# The values can be overridden by defining environment variables
+# If no value given, use this default:
+define_with_default BUILD_SDK false
+define_with_default BUILD_TEST_IMAGE false
+define_with_default COPY_LICENSES false
+define_with_default LAYER_ARCHIVE false
+define_with_default CREATE_RELEASE_DIR false
+define_with_default MIRROR "https://docs.projects.genivi.org/releases/yocto_mirror"
+define_with_default PREMIRROR ""  # By default none (but we have the shared DL_DIR)
+define_with_default RM_WORK false
+define_with_default REUSE_STANDARD_DL_DIR true
+define_with_default REUSE_STANDARD_SSTATE_DIR true
```

Comment:
I'd prefer the default for SSTATE_DIR to be false. 

It's easy to set this for the CI build and generates a fairly complex error message for whatever user right now.